### PR TITLE
Add support for aliases in ORDER BY.

### LINKED ISSUE #1220: TPC-H : Alias column is not referenced in an order by clause
# Feature Request
## Summary
For a column defined with an alias, we cannot use it in the same query in the order by.
Further investigation is required to understand the problem and possible solutions.

## Environment
I don't think it's an environment issue. But I'm running it on WSL2 on windows with Ubuntu 18.04.

## Steps to Reproduce
1. Compile with the following args: -DCMAKE_BUILD_TYPE=Release ..
2. Run terrier executable
3. Connect to the terrier terminal by executing : psql -h localhost -p 15721 -d terrier
4. Create the tables from "https://github.com/oltpbenchmark/oltpbench/blob/master/src/com/oltpbenchmark/benchmarks/tpch/ddls/tpch-postgres-ddl.sql" .
5. Run the query 

```SQL
select
        sum(c_acctbal * (1 - c_acctbal)) as revenue
from
        customer
group by
        c_name
order by
        revenue desc
limit 20;

Error:
 ERROR:  column "revenue" does not exist
```

### PR DESCRIPTION
**Pull request by gautam20197**
Fix #1220 

Changes in the PR:

1. Binder: Added binding of order by columns using alias by picking from the select list. If the alias does not exist in the select list, the binder will throw an error of the column not existing.

2. Optimizer: Create a derived value expression from children expression of the aggregate expression that will be sorted by the order by.

### CODE REVIEW

File: src/binder/bind_node_visitor.cpp [hunk 1]
```diff
@@ -516,6 +517,7 @@ void BindNodeVisitor::Visit(common::ManagedPointer<parser::SelectStatement> node
 
   if (node->GetSelectOrderBy() != nullptr) {
     UnifyOrderByExpression(node->GetSelectOrderBy(), node->GetSelectColumns());
+//    bool alias_unified = UnifyAliasNameOrderByExpression(node->GetSelectOrderBy(), node->GetSelectColumns());
```
Comment (lmwnshn) [inline src/binder/bind_node_visitor.cpp:520]: Dead code.

**Review by lmwnshn — CHANGES_REQUESTED** [review]
Could you remove the dead code? Will help you generate the trace files later.

File: src/binder/bind_node_visitor.cpp (hunk 1 above)
Comment (gautam20197) [reply src/binder/bind_node_visitor.cpp:520]: Yup I'll do that.

**Comment by lmwnshn** [comment]
Additionally, you may need to run format/tidy/etc first.

**Review by lmwnshn — APPROVED** [review]
LGTM if passes CI.

**Pull request merged**